pinot-core/src/main/java/org/apache/pinot/core/operator/combine/BaseCombineOperator.java

```diff
@@ -162,7 +162,7 @@ protected void processSegments() {
         if (operator instanceof AcquireReleaseColumnsSegmentOperator) {
           ((AcquireReleaseColumnsSegmentOperator) operator).acquire();
         }
-        resultsBlock = (T) operator.nextBlock();
+        resultsBlock = convertToMergeableBlock((BaseResultsBlock) operator.nextBlock());
```

Comment:
Good catch, but by simply changing `resultBlock` to `BaseResultsBlock` we cannot call `isQuerySatisfied` because it demands a `T`. I've applied some extra modifications, please check if they are good enough